lib/beaker/host.rb

```diff
@@ -300,7 +300,7 @@ def exec command, options={}
     # @param [String] dir The directory structure to create on the host
     # @return [Boolean] True, if directory construction succeeded, otherwise False
     def mkdir_p dir
-      if host['is_cygwin'].nil? or host['is_cygwin'] == true
+      if @options['is_cygwin'].nil? or @options['is_cygwin'] == true
```

Comment:
Looking over this again, we should use the host object inheritance to avoid doing an is_cygwin at all.  The 'mkdir -p' should be used by unix type hosts and the 'if not exists' should be pushed into the pswindows host.